app/models/manageiq/providers/vmware/infra_manager/refresh_parser_inventory_object.rb

```diff
@@ -82,7 +82,9 @@ def storage_inv_to_hashes(inv)
           )
         end
 
-        result << result.new_inventory_object(new_result)
+        new_result = result.new_inventory_object(new_result)
```

Comment:
Okay I'll try replacing all the result_uids, should clean things up a lot

> not sure what result_uids[:storage_id][uid] = new_result does though

Think that has something to do with the `mor` and the `uid` being different for storage, though I don't see `result_uids[:storage_id]` being used anywhere so maybe we can get rid of it